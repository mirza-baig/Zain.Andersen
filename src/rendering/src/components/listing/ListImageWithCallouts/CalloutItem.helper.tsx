import React from 'react';
import { Feature } from 'src/.generated/Feature.EnterpriseWeb.model';
import { BodyCopy } from 'src/helpers/BodyCopy';
import { Headline } from 'src/helpers/Headline';

type CalloutItemProps =
  Feature.EnterpriseWeb.Enterprise.Components.Listing.ListImageWithCallouts.ListImageWithCalloutsItem & {
    callOutItemClasses: {
      calloutContainer?: string;
      calloutItemHeadline?: string;
      calloutBody?: string;
    };
  };

const CalloutItem = ({ fields, callOutItemClasses }: CalloutItemProps) => {
  return (
    <div className={callOutItemClasses?.calloutContainer}>
      <Headline classes={callOutItemClasses?.calloutItemHeadline || ''} fields={fields} />
      <BodyCopy classes={callOutItemClasses?.calloutBody || ''} fields={fields} />
    </div>
  );
};

export default CalloutItem;
