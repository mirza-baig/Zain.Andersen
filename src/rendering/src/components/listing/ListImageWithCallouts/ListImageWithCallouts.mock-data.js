const defaultData = {
  rendering: {
    dataSource: 'sampledatasource',
    componentName: 'ListImageWithCallouts',
  },
  fields: {
    componentSpacing: null,
    sectionId: {
      value: '',
    },
    theme: {
      value: '',
    },
    headlineLevel: {
      id: '0f556f3a-bbad-4aa5-952d-b79003b39cd6',
      url: 'http://localhost/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/Headline-Levels/Heading-2',
      name: 'Heading 2',
      displayName: 'Heading 2',
      fields: {
        Value: {
          value: 'h2',
        },
      },
      templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
      templateName: 'Enum',
    },
    headlineText: {
      value:
        'List image with Callouts -headline Text with 3 items goes here List image with Callouts -headline Text with 3 items goes here',
    },
    primaryImage: {
      value: {
        src: 'https://source.unsplash.com/random/400x600?abstract',
        alt: '',
        width: '400',
        height: '600',
      },
    },
    primaryImageMobileFocusArea: {
      id: 'e6b36945-ebc0-4af8-ad95-d8268b56b7d0',
      url: 'http://localhost/sitecore/login/sitecore/system/Settings/Feature/EnterpriseWeb/Enums/Image-Focus/Center',
      name: 'Center',
      displayName: 'Center',
      fields: {
        Value: {
          value: 'center',
        },
      },
      templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
      templateName: 'Enum',
    },
    primaryImageMobile: {
      value: {
        src: 'https://source.unsplash.com/random/400x300?abstract',
        alt: '',
        width: '400',
        height: '300',
      },
    },
    imgPosition: {
      id: '870a817e-d7b9-41c7-a4c3-5c0008f2b2c1',
      url: 'http://localhost/sitecore/login/sitecore/system/Settings/Feature/EnterpriseWeb/Enums/Image-Position/Left',
      name: 'Left',
      displayName: 'Left',
      fields: {
        Value: {
          value: 'left',
        },
      },
      templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
      templateName: 'Enum',
    },
    backgroundColor: {
      id: '64333070-0d1b-4a33-9e19-7d299d6e8dd3',
      url: 'http://localhost/sitecore/login/sitecore/system/Settings/Feature/EnterpriseWeb/Enums/Background-Color/White',
      name: 'White',
      displayName: 'White',
      fields: {
        Value: {
          value: 'white',
        },
      },
      templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
      templateName: 'Enum',
    },
    eventName: {
      value: '',
    },
    eventType: {
      value: '',
    },
    eventZone: {
      value: '',
    },
    children: [
      {
        fields: {
          headlineLevel: {
            id: '0f556f3a-bbad-4aa5-952d-b79003b39cd6',
            url: 'http://localhost/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/Headline-Levels/Heading-2',
            name: 'Heading 2',
            displayName: 'Heading 2',
            fields: {
              Value: {
                value: 'h2',
              },
            },
            templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
            templateName: 'Enum',
          },
          headlineText: {
            value: 'List item 1',
          },
          body: {
            value:
              'Lorem ipsum is placeholder text commonly used in the graphic, print, and publishing industries for previewing layouts and visual mockups.<br />\n<div>&nbsp;</div>',
          },
        },
      },
      {
        fields: {
          headlineLevel: {
            id: '7f10288a-beb1-4d95-a139-32df40094441',
            url: 'http://localhost/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/Headline-Levels/Heading-3',
            name: 'Heading 3',
            displayName: 'Heading 3',
            fields: {
              Value: {
                value: 'h3',
              },
            },
            templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
            templateName: 'Enum',
          },
          headlineText: {
            value:
              'H3 List image with callouts item 2 goes here H3 List image with callouts item 2 goes here',
          },
          body: {
            value:
              'Lorem ipsum is placeholder text commonly used in the graphic, print, and publishing industries for previewing layouts and visual mockups.<br />\n<div>&nbsp;</div>',
          },
        },
      },
      {
        fields: {
          headlineLevel: {
            id: 'aada53f8-582e-4687-b9b6-3f2c01437a23',
            url: 'http://localhost/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/Headline-Levels/Heading-4',
            name: 'Heading 4',
            displayName: 'Heading 4',
            fields: {
              Value: {
                value: 'h4',
              },
            },
            templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
            templateName: 'Enum',
          },
          headlineText: {
            value:
              'H4 List image with callouts item 2 goes here H3 List image with callouts item 2 goes here',
          },
          body: {
            value:
              'Lorem ipsum is placeholder text commonly used in the graphic, print, and publishing industries for previewing layouts and visual mockups.<br />\n<div>&nbsp;</div>',
          },
        },
      },
      {
        fields: {
          headlineLevel: {
            id: 'aada53f8-582e-4687-b9b6-3f2c01437a23',
            url: 'http://localhost/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/Headline-Levels/Heading-4',
            name: 'Heading 4',
            displayName: 'Heading 4',
            fields: {
              Value: {
                value: 'h4',
              },
            },
            templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
            templateName: 'Enum',
          },
          headlineText: {
            value:
              'H4 List image with callouts item 2 goes here H3 List image with callouts item 2 goes here',
          },
          body: {
            value:
              'Lorem ipsum is placeholder text commonly used in the graphic, print, and publishing industries for previewing layouts and visual mockups.<br />\n<div>&nbsp;</div>',
          },
        },
      },
    ],
  },
};

export default defaultData;
