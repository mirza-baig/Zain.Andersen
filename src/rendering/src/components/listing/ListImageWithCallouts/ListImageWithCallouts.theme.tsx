import { ThemeFile } from 'lib/context/ThemeContext';

export const ListImageWithCalloutsTheme: ThemeFile = {
  aw: {
    classes: {
      listWrapper: 'pt-xl pb-xl',
      headline:
        'text-theme-text text-center md:text-left text-sm-m md:text-m font-bold mb-s md:mb-m',
      callOutItemClasses: {
        calloutContainer: 'mt-s md:mt-0 pt-s px-xs md:px-0 border-t border-theme-text',
        calloutItemHeadline: 'text-theme-text text-xs font-bold mb-xxs',
        calloutBody: 'text-theme-body text-body',
      },
      imageContainer: 'top-0',
    },
  },
  rba: {
    classes: {
      listWrapper: 'pt-m md:pt-l pb-m md:pb-l',
      headline: 'text-m md:text-l font-extra-light mb-xxs text-theme-text',
      callOutItemClasses: {
        calloutContainer: 'mt-s md:mt-0 pt-s px-xs md:px-0 border-t border-theme-text',
        calloutItemHeadline: '!font-serif text-xs font-bold mb-xxs text-theme-text uppercase',
        calloutBody: 'text-body text-theme-body',
      },
      imageContainer: 'top-0',
    },
  },
};
