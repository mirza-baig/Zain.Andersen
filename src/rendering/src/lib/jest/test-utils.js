// Global
import { render, fireEvent, waitFor } from '@testing-library/react';
import { ComponentPropsContext } from '@sitecore-jss/sitecore-jss-nextjs';
import { FastSitecoreContext } from 'lib/overrides/FastSitecoreContext';

// Lib
import { createSampleComponentFactory } from 'lib/mocks/mock-placeholder';

export const eeSitecoreContext = {
  route: {
    name: 'Foo',
    placeholders: {},
  },
  context: {
    pageEditing: true,
    site: {
      name: 'JssTestWeb',
    },
    language: 'en',
  },
};

export const renderComponent = (
  Component,
  {
    componentProps = {},
    staticProps = {},
    componentFactory = createSampleComponentFactory(),
    sitecoreContext = {
      route: {
        name: 'Foo',
        placeholders: {},
      },
    },
  }
) => {
  const mockLayoutData = {
    sitecore: sitecoreContext,
  };

  return render(
    <ComponentPropsContext value={staticProps}>
      <FastSitecoreContext componentFactory={componentFactory} layoutData={mockLayoutData}>
        <Component {...componentProps} />
      </FastSitecoreContext>
    </ComponentPropsContext>
  );
};

export const snapshot = (Component, mockData) => {
  const component = renderComponent(Component, mockData);
  expect(component.container.firstChild).toMatchSnapshot();
  return component;
};

export const hasDataComponent = async (component, name) => {
  await waitFor(() => {
    const nodes = component.baseElement.querySelectorAll(`[data-component="${name}"]`);
    return expect(nodes.length).toBe(1);
  });
};

export const mockDateAs = (epochMs) => {
  if (typeof epochMs !== 'number') {
    console.error(
      'mockDateAs only accepts epoch values in milliseconds to prevent timezones from factoring in to Date creation.'
    );
  }

  jest.spyOn(global.Date, 'now').mockImplementation(() => new Date(epochMs).valueOf());
};

export const testFieldError = async (Component, mockData, { label, value, error }) => {
  const component = renderComponent(Component, mockData);
  const field = component.getByLabelText(label);
  fireEvent.change(field, { target: { value } });
  field.blur();
  await waitFor(() => {
    const err = component.queryByText(error);
    return expect(err).not.toBe(null);
  });
};
