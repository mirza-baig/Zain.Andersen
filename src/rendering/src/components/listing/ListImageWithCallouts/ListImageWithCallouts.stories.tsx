// Global
import { Story, Meta } from '@storybook/react';
// Local
import ListImageWithCallouts, { ListImageWithCalloutsProps } from './ListImageWithCallouts';
import defaultData from './ListImageWithCallouts.mock-data';

export default {
  title: 'Listing/ListImageWithCallouts',
  component: ListImageWithCallouts,
} as Meta;
const Template: Story<ListImageWithCalloutsProps> = (props) => <ListImageWithCallouts {...props} />;
export const Default = Template.bind({});
Default.args = defaultData;
