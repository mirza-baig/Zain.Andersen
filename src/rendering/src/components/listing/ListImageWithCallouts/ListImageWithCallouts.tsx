// Components
import { withDatasourceCheck } from '@sitecore-jss/sitecore-jss-nextjs';
import classNames from 'classnames';
import { Feature } from 'src/.generated/Feature.EnterpriseWeb.model';
import { getEnum } from 'lib/utils';
import { ComponentProps } from 'lib/types/component-props';
import { useTheme } from 'lib/context/ThemeContext';
import { getBreakpoint, useCurrentScreenType } from 'lib/utils/get-screen-type';
import { Component } from 'src/helpers/Component';
import { Headline } from 'src/helpers/Headline';
import { ImagePrimary } from 'src/helpers/Media';
import { SliderWrapper } from 'src/helpers/SliderWrapper';
import CalloutItem from './CalloutItem.helper';
import { ListImageWithCalloutsTheme } from './ListImageWithCallouts.theme';

type CalloutItemProp =
  Feature.EnterpriseWeb.Enterprise.Components.Listing.ListImageWithCallouts.ListImageWithCalloutsItem;

export type ListImageWithCalloutsProps = {
  fields: Feature.EnterpriseWeb.Enterprise.Components.Listing.ListImageWithCallouts.ListImageWithCallouts['fields'] & {
    children: Array<CalloutItemProp>;
  };
} & ComponentProps;

const RowPositions: { [callOutIndex: number]: string } = {
  1: 'md:row-start-1',
  2: 'md:row-start-2',
  3: 'md:row-start-3',
};

const ColPositions: { [callOutIndex: number]: string } = {
  1: 'md:col-start-1',
  3: 'md:col-start-10',
};

const ListImageWithCallouts = (props: ListImageWithCalloutsProps): JSX.Element => {
  const { fields } = props;
  const { themeName, themeData } = useTheme(ListImageWithCalloutsTheme);
  const imagePosition = getEnum(fields?.imgPosition) || 'right';

  const { currentScreenWidth } = useCurrentScreenType();

  // We've to use this check in order remove optional chaining
  if (!fields) {
    return <></>;
  }

  // Callout items code
  const calloutItems = fields.children;

  const sliderSettings = {
    infinite: true,
    speed: 500,
    slidesToShow: 1,
    slidesToScroll: 1,
    enableNumberedPagination: themeName === 'rba',
  };

  const renderCallouts = () => {
    if (currentScreenWidth <= getBreakpoint('md')) {
      // Renderings for mobile devices
      return (
        <div
          className={classNames(
            'col-span-12',
            calloutItems.length > 3 ? 'md:col-span-3' : 'md:col-span-5'
          )}
        >
          <SliderWrapper sliderSettings={sliderSettings} theme={themeName}>
            {calloutItems.map((item: CalloutItemProp, index: number) => {
              return (
                <CalloutItem
                  key={index}
                  fields={item.fields}
                  callOutItemClasses={themeData.classes.callOutItemClasses}
                />
              );
            })}
          </SliderWrapper>
        </div>
      );
    } else {
      // Renderings for tablets and large screen devices
      return calloutItems.map(
        (
          item: Feature.EnterpriseWeb.Enterprise.Components.Listing.ListImageWithCallouts.ListImageWithCalloutsItem,
          index: number
        ) => {
          return (
            <div
              key={index}
              className={classNames(
                calloutItems.length > 3 ? 'md:col-span-3' : 'md:col-span-5',
                RowPositions[calloutItems.length === 4 ? (index % 2) + 2 : (index % 3) + 2],

                calloutItems.length > 3
                  ? calloutItems.length === 4
                    ? ColPositions[index > 1 ? 3 : 1]
                    : ColPositions[index > 2 ? 3 : 1]
                  : imagePosition === 'right'
                  ? 'md:col-start-1'
                  : 'md:col-start-7'
              )}
            >
              <CalloutItem
                fields={item.fields}
                callOutItemClasses={themeData.classes.callOutItemClasses}
              />
            </div>
          );
        }
      );
    }
  };

  return (
    <Component
      variant={themeName === 'aw' ? 'full' : 'lg'}
      backgroundVariant={getEnum(fields.backgroundColor)}
      dataComponent="listing/listimagewithcallouts"
      {...props}
    >
      <div className={classNames('col-span-12', themeData.classes.listWrapper)}>
        <div
          className={classNames(
            'grid-rows-auto grid grid-cols-12 gap-y-0 md:max-w-screen-lg md:grid-flow-row-dense md:grid-cols-12 md:gap-s md:px-m lg:mx-auto'
          )}
        >
          <div className="col-span-12">
            <Headline classes={themeData.classes.headline} {...props} />
          </div>
          <div
            className={classNames(
              'col-span-12 row-span-3 md:col-span-6 md:row-start-2',
              calloutItems.length > 3
                ? 'md:col-start-4'
                : imagePosition === 'right'
                ? 'md:col-start-7'
                : 'md:col-start-1',
              themeData.classes.imageContainer
            )}
          >
            <ImagePrimary imageLayout={'intrinsic'} ratio="portrait" focusArea={'top'} {...props} />
          </div>
          {renderCallouts()}
        </div>
      </div>
    </Component>
  );
};
export default withDatasourceCheck()(ListImageWithCallouts);
