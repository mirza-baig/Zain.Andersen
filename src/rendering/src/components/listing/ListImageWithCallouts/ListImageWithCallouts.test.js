// Global
import { hasDataComponent, snapshot } from 'lib/jest/test-utils';
import 'lib/jest/match-media-mock';
// Local
import ListImageWithCallouts from './ListImageWithCallouts';
import defaultData from './ListImageWithCallouts.mock-data';

it('renders correctly', async () => {
  const component = snapshot(ListImageWithCallouts, { componentProps: defaultData });
  await hasDataComponent(component, 'listing/listimagewithcallouts');
});
