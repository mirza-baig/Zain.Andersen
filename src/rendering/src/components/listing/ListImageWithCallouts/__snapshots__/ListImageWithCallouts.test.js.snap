// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  data-component="listing/listimagewithcallouts"
  id="id1695242607"
>
  <div
    class="my-8 w-full theme-white"
  >
    <div
      class="section-grid grid grid-cols-2 md:grid-cols-12 px-m gap-s md:gap-s "
    >
      <div
        class="col-span-12 pt-xl pb-xl"
      >
        <div
          class="grid-rows-auto grid grid-cols-12 gap-y-0 md:max-w-screen-lg md:grid-flow-row-dense md:grid-cols-12 md:gap-s md:px-m lg:mx-auto"
        >
          <div
            class="col-span-12"
          >
            <div
              class="items-top font-sans [&_a:hover]:underline text-theme-text text-center md:text-left text-sm-m md:text-m font-bold mb-s md:mb-m"
            >
              <h2>
                List image with Callouts -headline Text with 3 items goes here List image with Callouts -headline Text with 3 items goes here
              </h2>
            </div>
          </div>
          <div
            class="col-span-12 row-span-3 md:col-span-6 md:row-start-2 md:col-start-4 top-0"
          >
            <div
              class="relative hidden md:block  "
            >
              <span
                style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27400%27%20height=%27600%27/%3e"
                    style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                  />
                </span>
                <img
                  alt=""
                  data-nimg="intrinsic"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
                <noscript />
              </span>
            </div>
            <div
              class="relative block w-full md:hidden"
            >
              <span
                style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27400%27%20height=%27300%27/%3e"
                    style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                  />
                </span>
                <img
                  alt=""
                  data-nimg="intrinsic"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover; object-position: center;"
                />
                <noscript />
              </span>
            </div>
          </div>
          <div
            class="md:col-span-3 md:row-start-2 md:col-start-1"
          >
            <div
              class="mt-s md:mt-0 pt-s px-xs md:px-0 border-t border-theme-text"
            >
              <div
                class="items-top font-sans [&_a:hover]:underline text-theme-text text-xs font-bold mb-xxs"
              >
                <h2>
                  List item 1
                </h2>
              </div>
              <div
                class="text-theme-body text-body"
              >
                <div
                  class="body-copy"
                >
                  <div>
                    Lorem ipsum is placeholder text commonly used in the graphic, print, and publishing industries for previewing layouts and visual mockups.
                    <br />
                    

                    <div>
                       
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="md:col-span-3 md:row-start-3 md:col-start-1"
          >
            <div
              class="mt-s md:mt-0 pt-s px-xs md:px-0 border-t border-theme-text"
            >
              <div
                class="items-top font-sans [&_a:hover]:underline text-theme-text text-xs font-bold mb-xxs"
              >
                <h3>
                  H3 List image with callouts item 2 goes here H3 List image with callouts item 2 goes here
                </h3>
              </div>
              <div
                class="text-theme-body text-body"
              >
                <div
                  class="body-copy"
                >
                  <div>
                    Lorem ipsum is placeholder text commonly used in the graphic, print, and publishing industries for previewing layouts and visual mockups.
                    <br />
                    

                    <div>
                       
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="md:col-span-3 md:row-start-2 md:col-start-10"
          >
            <div
              class="mt-s md:mt-0 pt-s px-xs md:px-0 border-t border-theme-text"
            >
              <div
                class="items-top font-sans [&_a:hover]:underline text-theme-text text-xs font-bold mb-xxs"
              >
                <h4>
                  H4 List image with callouts item 2 goes here H3 List image with callouts item 2 goes here
                </h4>
              </div>
              <div
                class="text-theme-body text-body"
              >
                <div
                  class="body-copy"
                >
                  <div>
                    Lorem ipsum is placeholder text commonly used in the graphic, print, and publishing industries for previewing layouts and visual mockups.
                    <br />
                    

                    <div>
                       
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="md:col-span-3 md:row-start-3 md:col-start-10"
          >
            <div
              class="mt-s md:mt-0 pt-s px-xs md:px-0 border-t border-theme-text"
            >
              <div
                class="items-top font-sans [&_a:hover]:underline text-theme-text text-xs font-bold mb-xxs"
              >
                <h4>
                  H4 List image with callouts item 2 goes here H3 List image with callouts item 2 goes here
                </h4>
              </div>
              <div
                class="text-theme-body text-body"
              >
                <div
                  class="body-copy"
                >
                  <div>
                    Lorem ipsum is placeholder text commonly used in the graphic, print, and publishing industries for previewing layouts and visual mockups.
                    <br />
                    

                    <div>
                       
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;
